src/CommitmentsV2/SFA.DAS.CommitmentsV2.MessageHandlers/EventHandlers/OverlappingTrainingDateRequest/OverlappingTrainingDateEditEndDateEventHandler.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.Extensions.Logging;
+using NServiceBus;
+using SFA.DAS.CommitmentsV2.Configuration;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Data.Extensions;
+using SFA.DAS.CommitmentsV2.Messages.Commands;
+using SFA.DAS.CommitmentsV2.Models;
+using SFA.DAS.Encoding;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using SFA.DAS.CommitmentsV2.Types;
+
+namespace SFA.DAS.CommitmentsV2.Messages.Events.OverlappingTrainingDateRequest
+{
+    public class OverlappingTrainingDateEditEndDateEventHandler : IHandleMessages<OverlappingTrainingDateEvent>
+    {
+        private readonly ILogger<OverlappingTrainingDateEditEndDateEventHandler> _logger;
+        private readonly CommitmentsV2Configuration _commitmentsV2Configuration;
+        private readonly IEncodingService _encodingService;
+        private readonly Lazy<ProviderCommitmentsDbContext> _dbContext;
+        public OverlappingTrainingDateEditEndDateEventHandler(Lazy<ProviderCommitmentsDbContext> dbContext,
+            ILogger<OverlappingTrainingDateEditEndDateEventHandler> logger,
+            IEncodingService encodingService,
+            CommitmentsV2Configuration commitmentsV2Configuration)
+        {
+            _logger = logger;
+            _commitmentsV2Configuration = commitmentsV2Configuration;
+            _encodingService = encodingService;
+            _dbContext = dbContext;
+        }
+
+        public async Task Handle(OverlappingTrainingDateEvent message, IMessageHandlerContext context)
+        {
+            try
+            {
+                _logger.LogInformation($"Received {nameof(OverlappingTrainingDateEvent)} for Uln {message?.Uln}");
+
+                var apprenticeship = await _dbContext.Value.GetApprenticeshipAggregate(message.ApprenticeshipId, default);
+
+                if (apprenticeship.PaymentStatus == PaymentStatus.Completed)
+                {
+                    var sendEmailToEmployerCommand = BuildEmailToEmployerCommand(apprenticeship, message);
+
+                    await context.Send(sendEmailToEmployerCommand, new SendOptions());
+                }
+
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Send message to employer for Uln {message?.Uln}");
+                throw;
+            }
+        }
+
+        private SendEmailToEmployerCommand BuildEmailToEmployerCommand(Apprenticeship apprenticeship, OverlappingTrainingDateEvent message)
+        {
+
+            var sendEmailToEmployerCommand = new SendEmailToEmployerCommand(apprenticeship.Cohort.EmployerAccountId,
```

Comment:
The task requires the `FirstName` token to be set as the name of the specific user, not the organisation. If you take a look at the message handler for `SendEmailToEmployerCommand` then you will see that in order to populate the name you must specify the `NameToken` property of the `SendEmailToEmployerCommand`

Reply:
Done